src/FLOWVPM_timeintegration.jl

```diff
@@ -256,89 +270,104 @@ Steps the field forward in time by dt in a third-order low-storage Runge-Kutta
+                MM[2] = J[4]*G[1]+J[5]*G[2]+J[6]*G[3]
+                MM[3] = J[7]*G[1]+J[8]*G[2]+J[9]*G[3]
             else
                 # Classic scheme (Γ⋅∇)U
-                MM[1] = p.J[1,1]*p.Gamma[1]+p.J[1,2]*p.Gamma[2]+p.J[1,3]*p.Gamma[3]
-                MM[2] = p.J[2,1]*p.Gamma[1]+p.J[2,2]*p.Gamma[2]+p.J[2,3]*p.Gamma[3]
-                MM[3] = p.J[3,1]*p.Gamma[1]+p.J[3,2]*p.Gamma[2]+p.J[3,3]*p.Gamma[3]
+                MM[1] = J[1]*G[1]+J[4]*G[2]+J[7]*G[3]
+                MM[2] = J[2]*G[1]+J[5]*G[2]+J[8]*G[3]
+                MM[3] = J[3]*G[1]+J[6]*G[2]+J[9]*G[3]
             end
-
+            
             # Store Z under MM[4] with Z = [ (f+g)/(1+3f) * S⋅Γ - f/(1+3f) * Cϵ⋅Γ ] / mag(Γ)^2, and ϵ=(Eadv + Estr)/zeta_sgmp(0)
-            MM[4] = (f+g)/(1+3*f) * (MM[1]*p.Gamma[1] + MM[2]*p.Gamma[2] + MM[3]*p.Gamma[3])
-            MM[4] -= f/(1+3*f) * (C*get_SFS1(p)*p.Gamma[1] + C*get_SFS2(p)*p.Gamma[2] + C*get_SFS3(p)*p.Gamma[3]) * p.sigma[1]^3/zeta0
-            MM[4] /= p.Gamma[1]^2 + p.Gamma[2]^2 + p.Gamma[3]^2
+            MM[4] = (f+g)/(1+3*f) * (MM[1]*G[1] + MM[2]*G[2] + MM[3]*G[3])
+            MM[4] -= f/(1+3*f) * (C*get_SFS1(p)*G[1] + C*get_SFS2(p)*G[2] + C*get_SFS3(p)*G[3]) * get_sigma(p)[]^3/zeta0
+            MM[4] /= G[1]^2 + G[2]^2 + G[3]^2
 
             # Store qstr_i = a_i*qstr_{i-1} + ΔΓ,
             # with ΔΓ = Δt*( S - 3ZΓ - Cϵ )
-            p.M[1, 2] = a*p.M[1, 2] + dt*(MM[1] - 3*MM[4]*p.Gamma[1] - C*get_SFS1(p)*p.sigma[1]^3/zeta0)
-            p.M[2, 2] = a*p.M[2, 2] + dt*(MM[2] - 3*MM[4]*p.Gamma[2] - C*get_SFS2(p)*p.sigma[1]^3/zeta0)
-            p.M[3, 2] = a*p.M[3, 2] + dt*(MM[3] - 3*MM[4]*p.Gamma[3] - C*get_SFS3(p)*p.sigma[1]^3/zeta0)
+            M[4] = a*M[4] + dt*(MM[1] - 3*MM[4]*G[1] - C*get_SFS1(p)*get_sigma(p)[]^3/zeta0)
+            M[5] = a*M[5] + dt*(MM[2] - 3*MM[4]*G[2] - C*get_SFS2(p)*get_sigma(p)[]^3/zeta0)
+            M[6] = a*M[6] + dt*(MM[3] - 3*MM[4]*G[3] - C*get_SFS3(p)*get_sigma(p)[]^3/zeta0)
 
             # Store qsgm_i = a_i*qsgm_{i-1} + Δσ, with Δσ = -Δt*σ*Z
-            p.M[2, 3] = a*p.M[2, 3] - dt*( p.sigma[1] * MM[4] )
+            M[8] = a*M[8] - dt*( get_sigma(p)[] * MM[4] )
 
             # Update vectorial circulation
-            p.Gamma[1] += b*p.M[1, 2]
-            p.Gamma[2] += b*p.M[2, 2]
-            p.Gamma[3] += b*p.M[3, 2]
+            G[1] += b*M[4]
+            G[2] += b*M[5]
+            G[3] += b*M[6]
 
             # Update cross-sectional area
-            p.sigma[1] += b*p.M[2, 3]
+            get_sigma(p)[] += b*M[8]
 
-        end
+        end=#
 
         # Update the particle field: viscous diffusion
         viscousdiffusion(pfield, dt; aux1=a, aux2=b)
+        #println("tape entries after diffusion: $(length(ReverseDiff.tape(pfield.particles[1].X[1])) - l)")
+        #l = length(ReverseDiff.tape(pfield.particles[1].X[1]))
 
     end
 
-
+    # something here breaks ForwardDiff # will need to re-enable and make sure this works now.
```

Comment:
Is this resolved? @Eric-P-Green

Reply:
Yep. I think this is an old comment I added when narrowing down some AD bugs (that have long since been fixed)

Reply:
Awesome! Can you remove the obsolete comments and commit to `derivatives` and then resolve this?